mcon/parse.c

```diff
@@ -285,6 +285,54 @@ static int mongo_process_option(mongo_con_manager *manager, mongo_servers *serve
 }
 
 /* Option parser helpers */
+static int parse_auth_mechanism_properties(mongo_con_manager *manager, mongo_servers *servers, char *value, char **error_message)
+{
+	char *start, *end, *colon, *prop_name, *prop_value;
+
+	/* ignore empty value */
+	if (strlen(value) == 0) {
+		return 0;
+	}
+
+	start = value;
+
+	do {
+		end = strchr(start, ',');
+		colon = strchr(start, ':');
+
+		if (!colon) {
+			int len = strlen(start) + sizeof("Error while trying to parse auth mechanism properties: No separator for ''");
+
+			*error_message = malloc(len + 1);
+			snprintf(*error_message, len, "Error while trying to parse auth mechanism properties: No separator for '%s'", start);
+			return 3;
+		}
+
+		prop_name = mcon_strndup(start, colon - start);
+		prop_value = mcon_strndup(colon + 1, end - colon - 1);
+
+		if (strcasecmp(prop_name, "SERVICE_NAME") == 0) {
+			mongo_manager_log(manager, MLOG_PARSE, MLOG_INFO, "- Found auth mechanism property '%s': '%s'", prop_name, prop_value);
+			free(servers->options.gssapiServiceName);
```

Comment:
The option value is initialized to `strdup("mongodb")` and is only ever assigned to a newly allocated string thereafter. This was copied from the existing option parsing of `gssapiServiceName`.